app/voip/server/api/v1/Rooms.ts

```diff
@@ -0,0 +1,90 @@
+import { Meteor } from 'meteor/meteor';
+import { Match, check } from 'meteor/check';
+import { Random } from 'meteor/random';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+
+import { API } from '../../../../api/server';
+import { VoipRoom, LivechatVisitors } from '../../../../models/server/raw';
+import { LivechatVoip } from '../../../../../server/sdk';
+import { IVoipRoom, OmnichannelSourceType } from '../../../../../definition/IRoom';
+
+API.v1.addRoute('voip/room', {
+	async get() {
+		const defaultCheckParams = {
+			token: String,
+			rid: Match.Maybe(String),
+			agentId: Match.Maybe(String),
+		};
+		check(this.queryParams, defaultCheckParams);
+
+		const { token, rid: roomId, agentId } = this.queryParams;
+		const guest = await LivechatVisitors.getVisitorByToken(token, {});
+		if (!guest) {
+			throw new Meteor.Error('invalid-token');
+		}
+
+		let room;
+		if (!roomId) {
+			room = await VoipRoom.findOneOpenByVisitorToken(token, {});
+			if (room) {
+				return API.v1.success({ room, newRoom: false });
+			}
+			let agent;
+
+			let agentObj = null;
+			if (agentId) {
+				agentObj = await LivechatVoip.findAgent(agentId);
+			}
+			if (agentObj) {
+				const { username } = agentObj;
+				agent = { agentId, username };
+			}
+			const rid = Random.id();
+			const roomInfo = {
+				source: {
+					type: OmnichannelSourceType.API,
+				},
+			};
+			room = await LivechatVoip.getNewRoom(guest, agent, rid, roomInfo);
+			return API.v1.success({ room });
```

Comment:
Need to exclude protected fields from room before returning it

Reply:
Not sure how to do that. Lets discuss this in the meeting.